Modix.Bot/DiscordUserMessageTypeReader.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Discord;
+using Discord.Commands;
+using Microsoft.Extensions.DependencyInjection;
+using Modix.Data.Repositories;
+using Modix.Services.Core;
+
+namespace Modix
+{
+    public class DiscordUserMessageTypeReader : TypeReader
+    {
+        public override async Task<TypeReaderResult> ReadAsync(ICommandContext context, string input, IServiceProvider services)
+        {
+            if (ulong.TryParse(input, out var messageId))
+            {
+                var value = await FindMessageInUnknownChannelAsync(context, messageId, services);
+                var userMessage = value as IUserMessage;
+
+                if (userMessage is null)
+                    return TypeReaderResult.FromError(CommandError.ObjectNotFound, "Message not found.");
+
+                return TypeReaderResult.FromSuccess(new DiscordUserMessage(userMessage));
+            }
+
+            return TypeReaderResult.FromError(CommandError.ObjectNotFound, "Message not found.");
+        }
+
+        // TODO: extract to service layer after the PR that adds the DB check to the quote service version of this is merged.
+        private async Task<IMessage> FindMessageInUnknownChannelAsync(ICommandContext context, ulong messageId, IServiceProvider services)
```

Comment:
This seems like something we could want elsewhere, at some point. Can we move it to an `IMessageService` in `Modix.Services`?